### PR TITLE
Added VCF input support & bug fixes

### PR DESCRIPTION
**Pull request by PhilPalmer**
Added:
- Support for input VCF files (any SNPs with missing rsids will be annotated using bcftools/dbSNP)
- Fixed output VCF header to make it bcftools compatible
- Fixed bug with conversion of ancestry chromosome names
- Updated bcftools version/container used in the pipeline
- Other minor changes eg made the `mergeChromosomes ` process a bit cleaner

NB: currently only input VCFs aligned to the b37 reference are supported

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -9,22 +9,21 @@
...
+  params.genotypes_path.endsWith("vcf") || params.genotypes_path.endsWith("vcf.gz")
+}
 
 /*--------------------------------------------------
   Genotype input files from kindof 23AndMe style
 ---------------------------------------------------*/
 
 if(params.genotypes_path) {
-  if(params.genotypes_path.endsWith(".txt")) {
+  if(params.genotypes_path.endsWith(".txt") || isVcf() ) {
```
Comment (pprieto) [inline main.nf]: Can also turn this validation into a single function that checks at the same time for the valid input formats (txt and vcf files)

**Comment by pprieto** [comment]
Had a quick look, couldn't you actually add the annotation process regardless in the bcftools process which already generates VCF files ?

Is it correct that right now it will only annotate the variants if the input files are VCF files?

**Comment by PhilPalmer** [comment]
Yes, currently it will only annotate input VCF files. This could be added for other input files as well

**Comment by willgdjones** [comment]
Generally, 23andMe/Ancestry/ general DTC company files have annotations with them @PhilPalmer.

**Pull request merged**